prov/efa/src/efa_mr.c

```diff
@@ -315,7 +318,14 @@ static int efa_mr_reg_impl(struct efa_mr *efa_mr, uint64_t flags, void *attr)
 
 	efa_mr->mr_fid.mem_desc = efa_mr;
 	efa_mr->mr_fid.key = efa_mr->ibv_mr->rkey;
-	efa_mr->peer.iface = mr_attr->iface;
+	/*
+	 * Skipping the domain type check is okay here since util_domain is at
+	 * the beginning of efa_domain and rxr_domain.
+	 */
+	if (efa_mr->domain->util_domain.info_domain_caps & FI_HMEM)
+		efa_mr->peer.iface = mr_attr->iface;
+	else
+		efa_mr->peer.iface = FI_HMEM_SYSTEM;
 	if (mr_attr->iface == FI_HMEM_CUDA)
```

Comment:
I think you need to update this check to use efa_mr->peer.iface

Reply:
Thank you, fixed.